src/datasources/results/query-handlers/query-results/QueryResultsDataSource.test.ts

```diff
@@ -106,6 +107,8 @@ describe('QueryResultsDataSource', () => {
       });
 
     test('should throw error when API returns 429 status', async () => {
+        cleanup();
```

Comment:
Looking at the docs for this function: https://testing-library.com/docs/react-testing-library/api/#cleanup

It seems to indicate that `cleanup` is used for removing state of rendered React trees. I don't see any React trees rendered here. Is it clear why we need to clean React tree rendered in these tests?

If it is needed can we move it to the `afterEach` so that we can clean-up consistently?
Should we make sure to `cleanup` in afterEach for other tests consistently?